### PR TITLE
Documentation page for LNS module configuration

### PR DESCRIPTION
**Pull request by MaggieSalak**
# PR for issue #1582 

## What is being addressed

Configuration of the delay on LNS which doesn't own the device connection is documented.

## How is this addressed

Since the LNS configuration wasn't documented so far, a new documentation page has been added in the User guide, including the processing delay and all other LNS module settings which can be set via environment variables or module twin.

### CODE REVIEW

File: docs/user-guide/lns-module-configuration.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Network Server IoT Edge Module configuration
+
+The following table is providing a list of configuration options, to be provided as environment variables for manual configuration of the `LoRaWanNetworkSrvModule`:
+
+| Environment variable name   | Description                                                                            | Mandatory                           |
+| --------------------------- | -------------------------------------------------------------------------------------- | ----------------------------------- |
+| IOTEDGE_APIVERSION          | Version of IoT Edge; also used to determine if LNS is running as an IoT Edge module    | Yes, if running as IoT Edge module  |
+| IOTEDGE_IOTHUBHOSTNAME      | Host name of the IoT Hub                                                               | Yes                                 |
+| IOTEDGE_GATEWAYHOSTNAME     | Gateway host name                                                                      | Yes, if edgeHub gateway is enabled  |
```
Comment (danigian) [inline docs/user-guide/lns-module-configuration.md:9]: I would say that, unless someone is debugging locally, we can remove the IOTEDGE_ variables from this table as these are injected by IoT Edge runtime itself.

File: docs/user-guide/lns-module-configuration.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Network Server IoT Edge Module configuration
```
Comment (danigian) [inline docs/user-guide/lns-module-configuration.md:1]: Please have a look at the basics station configuration file header in order to remove the useless navigation sidebar on the right (useless as we don't have a TOC here and we could have more space for the table itself)

File: docs/user-guide/lns-module-configuration.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+| IOTEDGE_APIVERSION          | Version of IoT Edge; also used to determine if LNS is running as an IoT Edge module    | Yes, if running as IoT Edge module  |
+| IOTEDGE_IOTHUBHOSTNAME      | Host name of the IoT Hub                                                               | Yes                                 |
+| IOTEDGE_GATEWAYHOSTNAME     | Gateway host name                                                                      | Yes, if edgeHub gateway is enabled  |
+| ENABLE_GATEWAY              | Indicates whether the edgeHub gateway should be enabled or not                         | No (defaults to `true`)             |
+| IOTEDGE_DEVICEID            | The gateway deviceId                                                                   | Yes                                 |
+| HTTPS_PROXY                 | HTTPS proxy url                                                                        | No                                  |
+| RX2_DATR                    | RX2 data rate                                                                          | No (defaults to 1)                  |
```
Comment (danigian) [inline docs/user-guide/lns-module-configuration.md:13]: Useful to override the default regional RX2 Data Rate at a global level (defaults to null, meaning that we take the regional values)

File: docs/user-guide/lns-module-configuration.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+| RX2_FREQ                    | RX2 frequency                                                                          | No                                  |
```
Comment (danigian) [inline docs/user-guide/lns-module-configuration.md:14]: Useful to override the default regional RX2 Frequency at a global level (defaults to null, meaning that we take the regional values)

File: docs/user-guide/lns-module-configuration.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+| LOG_TO_CONSOLE              | Indicates whether logging to console is enabled or not                                 | No (default to `true`)              |
+| LOG_TO_TCP                  | Indicates whether logging to TCP is enabled or not (used mainly for integration tests) | No (defaults to `false`)            |
+| LOG_TO_HUB                  | Indicates whether logging to IoT Hub is enabled or not                                 | No (defaults to `false`)            |
+| LOG_TO_TCP_ADDRESS          | IP address for TCP logs                                                                | Yes, only if TCP logging is enabled |
+| LOG_TO_TCP_PORT             | TCP port to send logs to                                                               | Yes, only if TCP logging is enabled |
+| NETID                       | Gateway network ID                                                                     | No (defaults to network id 1)       |
+| AllowedDevAddresses         | Allowed Dev addresses                                                                  | No                                  |
```
Comment (danigian) [inline docs/user-guide/lns-module-configuration.md:25]: Sets a list of allowed dev addresses from which message processing is enabled.

Let's say in the default that not setting it allows all messages coming from the defined NETID to be processed

File: docs/user-guide/lns-module-configuration.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+| LNS_SERVER_PFX_PATH         | Path of the .pfx certificate to be used for LNS Server endpoint                        | No                                  |
+| LNS_SERVER_PFX_PASSWORD     | Password of the .pfx certificate to be used for LNS Server endpoint                    | No                                  |
+| CLIENT_CERTIFICATE_MODE     | Specifices the client certificate mode with which the server should be run             | No (defaults to `NoCertificate`)    |
```
Comment (danigian) [inline docs/user-guide/lns-module-configuration.md:28]:
```suggestion
| CLIENT_CERTIFICATE_MODE     | Specifies the client certificate mode with which the server should be run             | No (defaults to `NoCertificate`)    |
```

File: docs/user-guide/lns-module-configuration.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+| LNS_VERSION                 | Version of the LNS                                                                     | No                                  |
+| IOTHUB_CONNECTION_POOL_SIZE | IoT Hub connection pool size                                                           | No (defaults to 1)                  |
```
Comment (danigian) [inline docs/user-guide/lns-module-configuration.md:30]: Description: This sets the AMQP Connection Pool Size for communication to IoT Edge Hub / IoT Hub (depending on ENABLE_GATEWAY). Increasing this value to higher number will improve scalability (link to scalability documentation user-guide/scalability)

File: docs/user-guide/lns-module-configuration.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+Additional settings are available via desired properties of the Network Server module twin in IoT Hub:
+
+| Property name                 | Description                                                                                                                                   | Mandatory               |
+| ----------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------- |
+| FacadeServerUrl               | Azure Facade function url                                                                                                                     | Yes                     |
+| FacadeServerAuthCode          | Azure Facade function auth code                                                                                                               | Yes                     |
+| ProcessingDelayInMilliseconds | Processing delay (in milliseconds) which should be used for LNS that does not own connection for a device (applicable to multi-gateway setup) | No (defaults to 400 ms) |
```
Comment (danigian) [inline docs/user-guide/lns-module-configuration.md:38]: Let's change the description:
Processing delay (in milliseconds) to be used for the LNS not owning the connection for a device in a multi-gateway scenario (with link to scalability document)

**Review by danigian — CHANGES_REQUESTED** [review]

File: docs/user-guide/lns-module-configuration.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+| NETID                       | Gateway network ID                                                                     | No (defaults to network id 1)       |
+| AllowedDevAddresses         | Allowed Dev addresses                                                                  | No                                  |
+| LNS_SERVER_PFX_PATH         | Path of the .pfx certificate to be used for LNS Server endpoint                        | No                                  |
+| LNS_SERVER_PFX_PASSWORD     | Password of the .pfx certificate to be used for LNS Server endpoint                    | No                                  |
+| CLIENT_CERTIFICATE_MODE     | Specifices the client certificate mode with which the server should be run             | No (defaults to `NoCertificate`)    |
+| LNS_VERSION                 | Version of the LNS                                                                     | No                                  |
+| IOTHUB_CONNECTION_POOL_SIZE | IoT Hub connection pool size                                                           | No (defaults to 1)                  |
+
```
Comment (bastbu) [inline docs/user-guide/lns-module-configuration.md:31]: Should we also mention the `APPINSIGHTS_INSTRUMENTATIONKEY`?

File: docs/user-guide/lns-module-configuration.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+| IOTHUB_CONNECTION_POOL_SIZE | AMQP Connection Pool Size for communication to IoT Edge Hub / IoT Hub (depending on `ENABLE_GATEWAY`). Increasing this value to higher number will improve scalability; for more information see [Scalability](./scalability.md) | No (defaults to 1) |
+| APPINSIGHTS_INSTRUMENTATIONKEY | Instrumentation key for forwarding metrics to Application Insights                  | No                                  |
+
+The following settings can be configured via desired properties of the Network
+Server module twin in IoT Hub:
+
+| Property name                 | Description                                                                                                                                   | Mandatory               |
+| ----------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------- |
+| FacadeServerUrl               | Azure Facade function url                                                                                                                     | Yes                     |
```
Comment (MaggieSalak) [inline docs/user-guide/lns-module-configuration.md:42]: @danigian the function url (and the same for auth code) is essentially set both [through env variables](https://github.com/Azure/iotedge-lorawan-starterkit/blob/4b2cdf9d425c55ee0c67b5900fa8c9ff211c9206/LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/NetworkServerConfiguration.cs#L160) and [through the desired properties of the twin](https://github.com/Azure/iotedge-lorawan-starterkit/blob/4b2cdf9d425c55ee0c67b5900fa8c9ff211c9206/LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/ModuleConnection/ModuleConnectionHost.cs#L199). Do you know why this is the case?

File: docs/user-guide/lns-module-configuration.md [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+# Network Server IoT Edge Module configuration
+
+The following table is providing a list of configuration options, to be provided
+as environment variables for manual configuration of the
+`LoRaWanNetworkSrvModule`:
+
+| Environment variable name   | Description                                                                            | Mandatory                           |
+| --------------------------- | -------------------------------------------------------------------------------------- | ----------------------------------- |
+| ENABLE_GATEWAY              | Indicates whether the edgeHub gateway should be enabled or not                         | No (defaults to `true`)             |
+| IOTEDGE_DEVICEID            | The gateway deviceId                                                                   | Yes                                 |
```
Comment (danigian) [inline docs/user-guide/lns-module-configuration.md:15]:
```suggestion
```

File: docs/user-guide/lns-module-configuration.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+| HTTPS_PROXY                 | HTTPS proxy url                                                                        | No                                  |
+| RX2_DATR                    | RX2 data rate; useful to override the default regional RX2 Data Rate at a global level | No (defaults to null, regional default value is used) |
+| RX2_FREQ                    | RX2 frequency; useful to override the default regional RX2 Frequency at a global level | No (defaults to null, regional default value is used) |
+| IOTEDGE_TIMEOUT             | IoT Edge timeout in milliseconds                                                       | No (defaults to 0)                  |
```
Comment (danigian) [inline docs/user-guide/lns-module-configuration.md:19]:
```suggestion
```

File: docs/user-guide/lns-module-configuration.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+| FACADE_SERVER_URL           | Azure Facade function url                                                              | Yes                                 |
```
Comment (danigian) [inline docs/user-guide/lns-module-configuration.md:20]: Let's give an example "https://deployedfunction.azurewebsites.net/api"

File: docs/user-guide/lns-module-configuration.md (hunk 10 above)
Comment (Mandur) [reply docs/user-guide/lns-module-configuration.md:40]: I think this is in order to be able to run locally, we take this value from the env vars in this case

**Pull request merged**